### PR TITLE
ECER-3119: Registrant requirement flows

### PR DESCRIPTION
**Pull request by pcatkins**
## Description

- Links from ECE One Year registrant flow to ECE Five Years registrant flow
- Fixed bug in Registrant Card component
- Removed redundant ECE Assistant and ECE One Year registrant requirement components (same as new application requirements)
- Specialization selection from ECE Five Year requirements page to add SNE and/or ITE

## Related Jira Issue(s)

- [ECER-1173](https://eccbc.atlassian.net/browse/ECER-1173)
- [ECER-2628](https://eccbc.atlassian.net/browse/ECER-2628)
- [ECER-2631](https://eccbc.atlassian.net/browse/ECER-2631)
- [ECER-2632](https://eccbc.atlassian.net/browse/ECER-2632)
- [ECER-2627](https://eccbc.atlassian.net/browse/ECER-2627)

## Checklist
- [X] I have tested these changes locally.
- [ ] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)
Include screenshots to demonstrate the changes visually.

## Additional Comments (optional)
Any additional context or information that might be helpful for the reviewer.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/application.ts [hunk 1]
```diff
@@ -3,28 +3,48 @@ import { defineStore } from "pinia";
...
 
+import { useCertificationStore } from "./certification";
+import { useUserStore } from "./user";
 import { useWizardStore } from "./wizard";
 export interface ApplicationState {
   applications: Components.Schemas.Application[] | null | undefined;
   draftApplication: Components.Schemas.DraftApplication;
   application: Components.Schemas.Application | null;
 }
 export type ApplicationFlow =
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/application.ts:26]: @SoLetsDev, note change here building exhaustive list of application flows and getting the corresponding configuration file. I figured better do this all in one place...

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue [hunk 2]
```diff
@@ -109,61 +101,18 @@ export default defineComponent({
...
 
-    const latestCertificateExpiredMoreThan5Years = (dt1: string) => {
-      const dt2 = DateTime.fromISO(certificationStore.latestCertification?.expiryDate!);
-      const differenceInYears = Math.abs(DateTime.fromISO(dt1).diff(dt2, "years").years);
-      return differenceInYears > 5;
-    };
+    //Initialize wizard
+    await wizardStore.initializeWizard(applicationStore.applicationConfiguration, applicationStore.draftApplication);
 
-    const draftApplicationCreatedOn = applicationStore.draftApplication.createdOn || formatDate(DateTime.now().toString());
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue:131]: @SoLetsDev note this has moved to application store

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/SpecializedCertificationOptions.vue [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        label="Add Special Needs Educator (SNE)"
+        hide-details="auto"
+        @update:model-value="handleSelection"
+      ></v-checkbox>
+      <v-input
+        v-if="applicationStore.draftApplication.certificationTypes?.length == 0"
+        auto-hide="auto"
+        :model-value="selected"
+        :rules="[hasSelection]"
+      ></v-input>
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/SpecializedCertificationOptions.vue:46]: confused with the v-if on this component and the rules. 

We only check if the user has no certification type....

Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/SpecializedCertificationOptions.vue:46]: Yes, it's convoluted but we don't need a selection if the user has 5 Year and is being asked about adding Ite/Sne... But in the case of Specialization selection we need at least one to be selected.

**Review by SoLetsDev — APPROVED** [review]

**Pull request merged**